### ISSUE TITLE
Adding ink type tags to VipsImage, questions and troubles

### ISSUE THREAD

**Issue by angelmixu** [libvips/libvips#1555]
Hi!

As suggested in this PR ( https://github.com/libvips/libvips/pull/1489 ), I'm adding tags to VipsImage for getting information about channels/bands.

I'm using this branch on my cloned repo to do the changes, I have put some tags when loading PNG files, and when loading and savig TIFF files:
https://github.com/angelmixu/libvips/commit/cd52d0c3f2f418ec669f44c5de034e0af3c5144a

And some questions arised.

As this webpage suggests: https://www.awaresystems.be/imaging/tiff/tifftags/extrasamples.html
Extra channels denoting transparency should be set to EXTRASAMPLE_ASSOCALPHA, masks with EXTRASAMPLE_UNASSALPHA , and other unrelated data should be put as EXTRASAMPLE_UNSPECIFIED.

So, when opening a PNG file, I set the Alpha channel to VIPS_INK_TRANSPARENCY, and when saving this image to TIFF I put EXTRASAMPLE_ASSOCALPHA. But when opening in Photoshop the transparency seems to get whiter than what the PNG looks like.
On the other hand, if I set the alpha channel to EXTRASAMPLE_UNASSALPHA, Photoshop opens correctly the image. But the alpha channel should be set to EXTRASAMPLE_ASSOCALPHA as the webpage says, isn't it?

After some tests, I converted the PNG file to TIFF using Photoshop, and what Photoshop does is set the alpha channel to EXTRASAMPLE_ASSOCALPHA, but looks like it modifies the RGB channels, and loading the image looks correct.

Using for example Irfan View for loading files, looks like it's the other way around, so... I don't know if what I'm doing is correct or not.
Or maybe, there should be some premultiplication of the image when storing the EXTRASAMPLE_ASSOCALPHA channel?

Also, do you like this approach of setting ink information? Any suggestion/advice?

Thanks!

**Comment by jcupitt** [comment]
Yes, ASSOCALPHA means pre-multipled alpha, and you can only have one of them. libvips is not premultiplied, so you must use UNASSALPHA eveywhere.

**Comment by angelmixu** [comment]
Oh, I see.
Then I should save the transparency ink as UNASSALPHA, and the others to UNSPECIFIED, isn't it?
And when loading a tiff file with ASSOCALPHA, should I also set it to VIPS_INK_TRANSPARENCY, or there should be any additional transformation?

**Comment by jcupitt** [comment]
I think UNSPECIFIED is for unrelated information, so I would use UNASSALPHA everywhere (this is what libvips does).

ASSOCALPHA channels mean you must unpremultiply the pixel data using that channel -- tiff2vips has code to do this.
